pkg/api/handlers/libpod/pods.go

```diff
@@ -25,24 +31,75 @@ func PodCreate(w http.ResponseWriter, r *http.Request) {
 		runtime = r.Context().Value("runtime").(*libpod.Runtime)
 		err     error
 	)
-	var psg specgen.PodSpecGenerator
-	if err := json.NewDecoder(r.Body).Decode(&psg); err != nil {
-		utils.Error(w, "failed to decode specgen", http.StatusInternalServerError, errors.Wrap(err, "failed to decode specgen"))
+
+	opts := &entities.PodCreateOptions{Infra: true, NetFlags: &entities.NetFlags{}, Net: &entities.NetOptions{}}
+	buffer := bytes.NewBuffer(make([]byte, 0))
+	r.Body = ioutil.NopCloser(io.TeeReader(r.Body, buffer)) // create teereader so we can decode body twice, once for podOpts, once for Infra
```

Comment:
Why should the body be decoded twice?

Reply:
Infra is undergoing to a fundamental change which requires the usage of container cli options and pod create options. decoding twice ensures that both infra and the pod get the options they need since many pod create options are really just infra container options. Decoding twice is a pretty straightforward way of doing this.

Reply:
But why decode twice the pod create endpoint should accepts one common structure and unmarshal into this. Why unmarshal into two different structs.